packages/synapse-react-client/src/components/ChangePassword/ChangePassword.tsx

```diff
@@ -55,64 +33,18 @@ export default function ChangePassword() {
     if (newPassword !== confirmPassword) {
       displayToast('Passwords do not match.', 'danger')
     } else {
-      const changeRequest: ChangePasswordWithCurrentPassword = {
-        newPassword,
-        concreteType:
-          'org.sagebionetworks.repo.model.auth.ChangePasswordWithCurrentPassword',
-        username: userProfile?.userName!,
-        currentPassword: oldPassword,
-      }
-      changePassword(changeRequest)
-    }
-  }
-
-  function handleChangePasswordWithOtp(
-    code: string,
-    otpType: TwoFactorAuthOtpType,
-  ) {
-    if (twoFactorAuthErrorResponse) {
-      const changeRequest: ChangePasswordWithTwoFactorAuthToken = {
+      handleChangePasswordWithCurrentPassword(
+        userProfile?.userName!,
+        oldPassword,
         newPassword,
-        concreteType:
-          'org.sagebionetworks.repo.model.auth.ChangePasswordWithTwoFactorAuthToken',
-        userId: twoFactorAuthErrorResponse.userId,
-        twoFaToken: twoFactorAuthErrorResponse.twoFaToken,
-        otpType: otpType,
-        otpCode: code,
-      }
-      changePassword(changeRequest)
+      )
     }
   }
 
   return (
     <div>
-      {twoFactorAuthErrorResponse ? (
-        <>
-          {otpStep === 'VERIFICATION_CODE' && (
-            <Typography variant={'body1'} sx={{ my: 2 }} align={'center'}>
-              {TOTP_GUIDANCE_TEXT}
-            </Typography>
-          )}
-          {otpStep === 'RECOVERY_CODE' && (
-            <Typography variant={'body1'} sx={{ my: 2 }} align={'center'}>
-              {RECOVERY_CODE_GUIDANCE_TEXT_SHORT}
-            </Typography>
-          )}
-          <OneTimePasswordForm
-            step={otpStep}
-            onClickUseTOTP={() => {
-              setOtpStep('VERIFICATION_CODE')
-            }}
-            onClickUseBackupCode={() => {
-              setOtpStep('RECOVERY_CODE')
-            }}
-            isLoading={isLoadingUserProfile || changePasswordIsPending}
-            onSubmit={handleChangePasswordWithOtp}
-          />
-          <Alert severity={'info'} sx={{ my: 2 }}>
-            Two-factor authentication is required to change your password.
-          </Alert>
-        </>
+      {promptForTwoFactorAuth ? (
+        twoFactorAuthPrompt
```

Comment:
Use the 2fa prompt returned by the custom hook, so both components have a common UI